File: my-folder/0238-product-of-array-except-self/solution.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
       int lengthNewArr=nums.length;
       int[] numArr=new int[nums.length];
      int product=1;
       for(int i=0;i<nums.length;i++)
       {
           numArr[i]=product;
           product=product*nums[i];
         
       }
        
       product=1;
       for(int i=lengthNewArr-1;i>=0;i--)
       {
           numArr[i]=numArr[i]*product;
           product=product*nums[i];
       }
        
        return numArr;
    }
    
 
}

File: my-folder/0238-product-of-array-except-self/solution.kt
class Solution {
    fun productExceptSelf(nums: IntArray): IntArray {
        var prefix=1
        val result=IntArray(nums.size)
        for(i in nums.indices)
        {
            result[i]=prefix
            prefix=prefix*nums[i]
        }
        var suffix=1
        for(i in nums.indices.reversed())
        {
            result[i]=result[i]*suffix
            suffix=suffix*nums[i]
        }
        return result
    }
}
